SS_Activation_OpenMC.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb  8 08:16:12 2024
+
+@author: Anupama Rajendra
+"""
+import openmc
+import openmc.deplete
+import numpy as np
+from fractions import Fraction
+import matplotlib.pyplot as plt
+
+openmc.config['chain_file'] = 'chain_endfb71_sfr.xml'
+
+# Create materials & export to XML:
+#Simulating tungsten shell:
+W = openmc.Material(material_id=1, name='W_Shell')
+W.set_density('g/cm3', 19.28)
+W.add_element('W', 1.00)
+materials = openmc.Materials([W])
+materials.export_to_xml()
+
+# Create geometry
+#Spherical shell:
+    
+R_1 = 1000
+S_1= openmc.Sphere(r=R_1) #sphere of radius 1000cm
+inside_sphere_1 = -S_1
+outside_sphere_1 = +S_1
+R_2 = 1005
+S_2 = openmc.Sphere(r=R_2, boundary_type='vacuum')
+inside_sphere_2 = -S_2
+outside_sphere_2 = +S_2
+S_3 = outside_sphere_1 & inside_sphere_2 #filled with tungsten
+
+# Mapping materials to geometry:
+Void = openmc.Cell(fill=None, region = inside_sphere_1)
+Shell = openmc.Cell(fill=W, region=S_3)
+geometry = openmc.Geometry([Void, Shell])
+geometry.export_to_xml()
+
+# Source distribution:
+PointSource = openmc.stats.Point(xyz=(0.0, 0.0, 0.0))
+Prob = openmc.stats.Discrete(14E+06, 1.0)
+
+# Assign simulation settings
+settings = openmc.Settings()
+settings.batches = 10
+settings.inactive = 1
+settings.particles = 10000
+settings.source = openmc.Source(space=PointSource, energy=Prob, strength = 1.0, particle = 'neutron')
+settings.run_mode = 'fixed source'
+settings.export_to_xml()
+
+# Define tallies
+neutron_tally = openmc.Tally(name="Neutron tally")
+neutron_tally.scores = ['flux', 'elastic', 'absorption']
+# Implementing filter for neutron tally through W shell
+cell_filter = openmc.CellFilter([Shell])
+neutron_tally.filters = [cell_filter]
+
+# Creating a tally to get the flux energy spectrum.
+# An energy filter is created to assign to the flux tally.
+e_min_1, e_max_1 = 5e2, 14.001e6
+groups = 500
+energies = np.logspace(np.log10(e_min_1), np.log10(e_max_1), groups + 1)
+energy_filter_flux = openmc.EnergyFilter(energies)
+
+e_min_2, e_max_2 = 5e2, 14.001e6
+groups = 500
+energies_2 = np.logspace(np.log10(e_min_2), np.log10(e_max_2), groups + 1)
+energy_filter_ng = openmc.EnergyFilter(energies_2)
+
+spectrum_tally = openmc.Tally(name="Flux spectrum")
+# Implementing energy and cell filters for flux spectrum tally
+spectrum_tally.filters = [energy_filter_flux, cell_filter]
+spectrum_tally.scores = ['flux']
```

Comment:
I think I was looking for a neutron flux as a function of energy and just a total elastic/absorption score.

Reply:
Yes, it can provide a total tally in addition to the spectrum, but both have to be asked for explicitly. The total values should be the same as in PR #3 , but the spectrum may be different as we're accounting for Vitamin-J now